schema/ext/vulnerability-1.0.xsd

```diff
@@ -0,0 +1,274 @@
+                                    example)
+                                </xs:documentation>
+                            </xs:annotation>
+                        </xs:element>
+                    </xs:sequence>
+                </xs:complexType>
+            </xs:element>
+            <xs:element name="severity" type="vuln:severityType" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="source" type="vuln:scoreSourceType" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="vector" type="xs:string" minOccurs="0" maxOccurs="1">
+                <xs:annotation>
+                    <xs:documentation xml:lang="en">
+                        Textual representation of the metric values used to score the vulnerability
+                        see attack vector in https://www.first.org/cvss/v3.1/specification-document
+                    </xs:documentation>
+                </xs:annotation>
+            </xs:element>
+        </xs:sequence>
+    </xs:complexType>
+
+    <xs:complexType name="vulnerability">
+        <xs:annotation>
+            <xs:documentation xml:lang="en">
+                Defines the structure of a vulnerability.
+            </xs:documentation>
+        </xs:annotation>
+        <xs:sequence>
+            <xs:element name="id" type="xs:normalizedString" minOccurs="1" maxOccurs="1">
+                <xs:annotation>
+                    <xs:documentation xml:lang="en">
+                        The id of the vulnerability as defined by the risk scoring methodology
+                        For example CVE-2019-15842 (of https://nvd.nist.gov/vuln/detail/CVE-2019-15842)
+                    </xs:documentation>
+                </xs:annotation>
+            </xs:element>
+            <xs:element name="source" minOccurs="0" maxOccurs="1">
+                <xs:complexType>
+                    <xs:annotation>
+                        <xs:documentation xml:lang="en">
+                            The source of the vulnerability where it is documented.
+                            Usually the name of the organization publishing vulnerability information
+                        </xs:documentation>
+                    </xs:annotation>
+                    <xs:sequence>
+                        <xs:element name="url" minOccurs="0" maxOccurs="1">
+                            <xs:annotation>
+                                <xs:documentation xml:lang="en">
+                                    The url of the vulnerability documentation as provided by the source
+                                    For example https://nvd.nist.gov/vuln/detail/CVE-2019-15842
+                                </xs:documentation>
+                            </xs:annotation>
+                            <xs:simpleType>
+                                <xs:restriction base="xs:string">
+                                    <xs:pattern value="(^(([^:/?#]+):)?(//([^/?#]*))?([^?#]*)(\?([^#]*))?(#(.*))?)"/>
```

Comment:
In the doc above, https://nvd.nist.gov/vuln/detail/CVE-2019-15842 does nit validate against the pattern specified. Regex in XML Schema is very limited - not all features are supported. You may have to work within those limitations, correct the expression, or use anyURI instead. URI obviously not as strict as URL.

Reply:
> In the doc above, https://nvd.nist.gov/vuln/detail/CVE-2019-15842 does not validate against the pattern specified. Regex in XML Schema is very limited - not all features are supported. You may have to work within those limitations, correct the expression, or use anyURI instead. URI obviously not as strict as URL.

Yup, `anyURI` seems a better fit. 35bb842